src/hal/src/command/graphics.rs

```diff
@@ -315,7 +326,7 @@ impl<'a, B: Backend, C: Supports<Graphics>, S: Shot> CommandBuffer<'a, B, C, S,
     ) -> RenderPassInlineEncoder<B, Primary>
     where
         T: IntoIterator,
-        T::Item: Borrow<ClearValue>,
+        T::Item: Borrow<BeginRenderPassClear>,
```

Comment:
do we really need the `Borrow` here? I would prefer `T: Iterator<Item = (AttachmentId, ClearValue)>` instead

Reply:
I don't think it's needed, but every function that needs an iterator has it, a trait bound of Borrow inside a IntoIterator. It seemed to me that this is a pattern we need to follow.

Reply:
`Borrow` is needed for immutable stuff, e.g. if you need a list of `B::Buffer`, it's not reasonable to accept an iterator over them, since the user has them wrapped into something at all times. For the clear colors, they are relatively light and copy-able, so there is no need for borrow.

Reply:
agree, it will be `T: Iterator<Item = (AttachmentId, ClearValue)>`

Reply:
@kvark  The reason that these were `IntoIterator<Item = Borrow<>>` initially is because arrays don't have an `into_iter` method and so users would have to type `&[clear1, clear2].iter().cloned()` whenever they used this method. However, I haven't looked but if this method needs to use the value of these variables rather than a borrow then this isn't helpful. Although it might still be sensible to require an `IntoIterator` over an `Iterator`.

Reply:
I had to call `.iter().cloned()` when specifying the clear values.

You can call `into_iter` on an array, but it will give `&T` instead of `T`, just like calling `into_iter` on a slice.

Btw if the parameter was just a slice, then the sorting wouldn't be needed and it wourld be possible to use `.iter().position()` instead of sorting.